### PR TITLE
Support for Doctrine/ORM 3

### PR DESCRIPTION
**Pull request by paranoiq**

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -1,5 +1,3 @@
-* text=auto
```
Comment (paranoiq) [inline .gitattributes:1]: causes git to use autocrlf=true, which rewrites all newlines to \r\n and cs and ec checks won't pass

File: tests/WithEntityManagerTest.php [hunk 2]
```diff
@@ -20,6 +21,7 @@ public function createEntityManager(): EntityManagerInterface
         $databaseFile = $tmpDir . '/db.sqlite';
 
         if (is_file($databaseFile)) {
+            gc_collect_cycles();
```
Comment (paranoiq) [inline tests/WithEntityManagerTest.php:28]: cannot delete open file on win : E

File: .gitignore [hunk 3]
```diff
@@ -5,3 +5,4 @@
 !/tmp/.gitkeep
 /.idea
 /phpstan.neon
+/composer.lock
```
[the remark this answers is not in the document]
Comment (janedbal) [reply .gitignore:8]: So if we drop it, we need at least dev deps to be pinned in composer.json so that the build is stable

File: composer.json [hunk 4]
```diff
@@ -5,21 +5,22 @@
...
+        "doctrine/orm": "^3.0.0",
+        "symfony/console": "^5.4.0 || ^6.0.0 || ^7.0.0"
     },
     "require-dev": {
         "doctrine/annotations": "^2.0.0",
-        "editorconfig-checker/editorconfig-checker": "^10.3.0",
+        "editorconfig-checker/editorconfig-checker": "^10.4.0",
         "ergebnis/composer-normalize": "^2.28",
-        "phpstan/phpstan": "^1.9.14",
+        "phpstan/phpstan": "^1.10.0",
```
Comment (janedbal) [inline composer.json:17]: I think we can go with latest 1.11

File: composer.json [hunk 5]
```diff
@@ -5,21 +5,22 @@
...
         "phpstan/phpstan-phpunit": "^1.3.3",
         "phpstan/phpstan-strict-rules": "^1.4.5",
         "phpunit/phpunit": "^9.5.28",
+        "psr/log": "^1 || ^2 || ^3",
         "shipmonk/composer-dependency-analyser": "^1.0",
-        "shipmonk/phpstan-rules": "^2.2.0",
+        "shipmonk/phpstan-rules": "^2.5.0",
```
Comment (janedbal) [inline composer.json:23]: I think we can go with latest 3.0.0

File: composer.json [hunk 6]
```diff
@@ -5,22 +5,23 @@
         "MIT"
     ],
     "require": {
-        "php": "^7.4 || ^8.0",
-        "doctrine/dbal": "^2.13.1 || ^3.1.4",
-        "doctrine/orm": "^2.7.0",
-        "symfony/console": "^4.0.0 || ^5.0.0 || ^6.0.0"
+        "php": "^8.1",
+        "doctrine/dbal": "^3.6.0",
```
Comment (janedbal) [inline composer.json:9]: Can we go `^3.6.0 || ^4.0.0` ?

Comment (janedbal) [reply composer.json:9]: Done in followup MR.

**Pull request merged**